text/0811-element-modifiers.md

```diff
@@ -0,0 +1,232 @@
+that complexity comes along with much more *control*. It also allows us to hook
+into Ember's dependency injection system, for example to use services.
+
+[RFC #432: Contextual Helpers and Modifiers][rfc-0432] allows optional
+invocation and partial application of modifiers, and allows modifiers to be
+yielded, passed as arguments, etc. [RFC #779: First-Class Component
+Templates][rfc-0779] allows defining them locally or importing them directly to
+be used in a `<template>` context. This combination makes many patterns much
+easier to implement:
+
+```js
+// /app/components/slide-up-card.js
+import Component from '@glimmer/component';
+import { modifier } from 'ember-modifier';
+
+const lockBodyScroll = modifier((element, [shouldLockBodyScroll]) => {
+  document.body.classList.add('lock-scroll');
+  return () => document.body.classList.remove('lock-scroll');
+});
+
+export default class SlideUpCard extends Component {
+  @action closeCard (event) {
+    event.preventDefault();
+    this.args.close();
+  }
+
+  <template>
+    {{#if has-block}}
+      {{yield (hash lock=lockBodyScroll onClick=this.closeCard)}}
+    {{else}}
+      <div
+        class="slide-up-card"
+        ...attributes
+        {{lockBodyScroll}}
+      >
+        <p>{{@bodyContent}}</p>
+
+        <button {{on "click" this.closeCard}}>Dismiss</button>
+      </div>
+    {{/if}}
+  </template>
+}
+```
+
+## Detailed design
+
+The necessary changes to `ember-cli` are relatively small since we only need
+to add the dependency to the `app` blueprint, and the `addon` blueprint will
+inherit it automatically.
+
+This has the advantage (over including it as an implicit dependency), that
+apps and addons that don't want to use it for some reason can opt out by
+removing the dependency from their `package.json` file.
+
+## How we teach this
+
+Ember Guides already teach how to use modifiers in "Template Lifecycle, DOM, and Modifiers"
+section of the guides.
+
+However, there are a couple changes we need to make to the content of the
+guides as they stand today:
+
+1. We will remove the "install `ember-modifier`" instruction from the guides,
+   since it will already be part of the blueprint.
+2. We should include at least a minimal example of a class-based modifier. For
+   that example, we can use something like `setInterval` to show how setup,
+   updates, and teardown can be more convenient when using a class.
+   (Note that here we should *only* teach the [newly-redesigned API][v3.2.0], not
+   the deprecated previous APIs. This will simplify our teaching significantly,
+   since we can describe how it matches the `Helper` APIs.)
+
+## Drawbacks
+
+- If we merge [RFC #757 "Default Modifier Manager][rfc-0757],
+  it may seem redundant with this RFC.
+- There is the potential for confusion between the `ember-modifier` package and
+  the `@ember/modifier` package which ships as part of `ember-source`.
+
+## Alternatives
+
+- Introduce [RFC #757 "Default Modifier Manager][rfc-0757]
```

Comment:
I think this is the direction we should go in. [ember-modifier's function-based modifiers](https://github.com/ember-modifier/ember-modifier#function-based-modifiers) feel like a stop-gap solution for the default modifier manager and it would be better to simply focus on that RFC instead. It would also be consistent with the already merged ["Default helper manager" RFC](https://github.com/emberjs/rfcs/pull/756) and is one less thing we would have to import.

I realize adding the `modifier` function would be the modifier equivalent of the already existing [`helper` function](https://api.emberjs.com/ember/4.3/functions/@ember%2Fcomponent%2Fhelper/helper), but I think that should be deprecated once the default helper manager is shipped :smile:.

The scope of this RFC could then be reduced to only the class based modifier API of ember-modifiers which ideally, as mentioned in the RFC and by others, would be exported from `@glimmer/modifier` or `@ember/modifier`.

Thanks for writing this RFC :heart:!